torchinductor/codegen/triton.py

```diff
@@ -200,13 +200,25 @@ def floor(x):
             tmp = ops.trunc(x)
             return f"tl.where({tmp}>{x}, {tmp}-1, {tmp})"
 
+    @staticmethod
+    def floordiv(a, b):
+        # See the comment in lowering.div_mode
+        div = f"tl.fdiv({a}.to(tl.float32), {b}.to(tl.float32), True)"
+        return f"{ops.floor(div)}"
+
     @staticmethod
     def trunc(x):
         if has_triton_libdevice():
             return f"tl.libdevice.trunc({x})"
         else:
             return f"{x}.to(tl.int32).to(tl.float32)"
 
+    @staticmethod
+    def truncdiv(a, b):
+        # See the comment in lowering.div_mode
+        div = f"tl.fdiv({a}.to(tl.float32), {b}.to(tl.float32), True)"
+        return f"{ops.trunc(div)}"
```

Comment:
```suggestion
        return ops.trunc(div)
```

Reply:
Converting integer numbers to float32 is wrong (you are losing precision), and, since you only ever send integer arguments here, you can just do return f"{{a} // {b}}, triton supports that for integral args.
Please add tests that would catch that precision loss from fp32, you'd need numbers more than 2**24 for this.
Note (this is confusing), triton's `//` is actually `trunc` mode, equivalent to c-style integer division, not to python `//`

Reply:
No wonder my earlier try of using `//` for floordiv didn't work. Isn't this a triton bug then, given triton still lives in the Python world?

Reply:
Yeah that's triton bug, we should file triton issue for that.

Reply:
https://github.com/openai/triton/issues/619